### PR TITLE
Use new Ruby and Nodejs versions

### PR DESCRIPTION
**Pull request by bkmgit**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -56,7 +55,8 @@ jobs:
           echo "ORIGIN=${owner}/${repository}" >> ${GITHUB_ENV}
           echo "TARGET_BRANCH=gh-pages" >> ${GITHUB_ENV}
           # "url:" is for the opengraph tags, and it can't be relative
-          echo "url: https://${owner}.github.io/${repository}" >> _extra_config.yml
+          echo "url: https://${owner}.github.io" >> _extra_config.yml
+          echo "baseurl: /${repository}" >> _extra_config.yml
```
Comment (kou) [inline .github/workflows/deploy.yml:59]: We have `JEKYLL_BASE_URL` for it: https://github.com/apache/arrow-site/blob/90107a822a97f69da6b1999fd44bd9dfa2ab8bee/Rakefile#L54-L55

Why do we need this?

Comment (bkmgit) [reply .github/workflows/deploy.yml]: Thanks. Updated.  Preview at: 
https://bkmgit.github.io/arrow-site/
due to
https://github.com/bkmgit/arrow-site/pull/4

**Comment by bkmgit** [comment]
Closes https://github.com/apache/arrow-site/issues/456

**Comment by bkmgit** [comment]
It seems to build. This is not needed.
https://github.com/bkmgit/arrow-site/pull/1

**Pull request closed without merging**